src/pages/404.tsx

```diff
@@ -1,13 +1,36 @@
+import { useLocation } from "@gatsbyjs/reach-router"
 import { PageProps } from "gatsby"
+import { useMemo } from "react"
+import Layout from "../components/Layout"
+import SideBar from "../components/SideBar"
+import { PageContext, PageInfo } from "../contexts/PageContext"
+import { pathWithOtherLocale } from "../paths"
 
-const NotFoundPage: React.FC<PageProps> = (props: PageProps) => {
-  const { path } = props
-
+const NotFoundPage: React.FC<PageProps> = () => {
+  const { pathname } = useLocation()
+  const locale = pathname.startsWith("/en/") ? "en" : "fi"
+  const context: PageInfo = useMemo(
+    () => ({ locale, localeLink: pathWithOtherLocale("", locale) }),
+    []
+  )
   return (
-    <main>
-      <h1>Not found</h1>
-      <p>{path}</p>
-    </main>
+    <PageContext.Provider value={context}>
+      <Layout>
+        <SideBar>
+          {locale === "fi" ? (
+            <>
+              <h1>Sivua {pathname} ei löytynyt</h1>
```

Comment:
We should change this to `<H1>` from typography.tsx when merging (or rebase/merge it to be so beforehand)